numpy/core/memmap.py

```diff
@@ -209,10 +209,10 @@ def __new__(subtype, filename, dtype=uint8, mode='r+', offset=0,
         import os.path
         try:
             mode = mode_equivalents[mode]
-        except KeyError:
+        except KeyError as e:
             if mode not in valid_filemodes:
-                raise ValueError("mode must be one of %s" %
-                                 (valid_filemodes + list(mode_equivalents.keys())))
+                raise ValueError("mode must be one of %s. Is currently %s." %
```

Comment:
More typical would be to use `(got ...)` I think, Using `%r` adds the quotes too.
```suggestion
                raise ValueError("mode must be one of %s (got %r)" %
```